### DISCUSSION TITLE
Indicating a dev resource in the AppHost

### DISCUSSION THREAD

**Discussion by normj** [microsoft/aspire#900] [Q&A]
Seems like the intention is all of the configurations setup in the AppHost will be written to the manifest for other tools to handle deployment. Some of the settings setup in the AppHost might be purely for development purposes. For example Hosting components that environment variables on project resources to emulators.

Has there been discussion on how to indicate something done in the AppHost is purely for local development purposes and not for deployment tooling?

**Comment by normj** [comment]
The emulator example might not be the best because my emulator resources could skip writing annotations which I think would skip it from getting in the manifest. But If I add a Redis container in the AppHost when I deploy to the cloud I would probably want to use the cloud provider's Redis service instead of my own Redis container.

**Reply by davidfowl** [reply]
The way we accomplish this is by having the tool that interprets the manifest translate the redis resource type as a the cloud provider’s natvie service type. So imagine there was an aws aspire deploy command, that would interpret the manifest and interpret redis as Amazon’s redis offering.

**Reply by davidfowl** [reply]
Here's how azd handles the manifest https://github.com/Azure/azure-dev/blob/main/cli/azd/pkg/apphost/generate.go#L215
